ts/src/profiler.ts

```diff
@@ -13,47 +13,323 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import * as delay from 'delay';
+import * as http from 'http';
 import * as path from 'path';
+import * as pify from 'pify';
+import * as zlib from 'zlib';
 
-import {AuthenticationConfig, Common, Logger, Service, ServiceConfig} from '../third_party/types/common-types';
-
-import {Config, ProfilerConfig} from './config';
+import {perftools} from '../../proto/profile';
+import {AuthenticationConfig, Common, Logger, Service, ServiceConfig, ServiceObject, ServiceObjectConfig} from '../third_party/types/common-types';
+import {ProfilerConfig} from './config';
 import {HeapProfiler} from './profilers/heap-profiler';
 import {TimeProfiler} from './profilers/time-profiler';
 
+export const common: Common = require('@google-cloud/common');
 const pjson = require('../../package.json');
-const common: Common = require('@google-cloud/common');
+const API = 'https://cloudprofiler.googleapis.com/v2';
+const zoneNameLabel = 'zone';
+const versionLabel = 'version';
+const instanceLabelName = 'instance';
+const scope = 'https://www.googleapis.com/auth/monitoring.write';
+const gzip = pify(zlib.gzip);
+
+enum ProfileTypes {
+  Wall = 'WALL',
+  Heap = 'HEAP'
+}
+
+/**
+ * Returns true if http status code indicates an error.
+ */
+function isErrorResponseStatusCode(code: number) {
+  return code < 200 || code >= 300;
+}
+
+/**
+ * Returns true if http status code indicates request should be retried.
+ */
+function isRetriableResponseStatusCode(code: number) {
+  // TODO: determine which codes one should not retry on.
```

Comment:
For this and the isRetriableError functions, I feel we should either figure the conditions in this pull request, or remove the associated logic and just always retry for now. Having these functions doesn't serve much purpose right now, and I would be worried that the code stays this way for long time and it's a confusing state.

Reply:
Removing these for now.